packages/@uppy/core/src/Uppy.test.js

```diff
@@ -702,6 +702,29 @@ describe('src/Core', () => {
       expect(onBeforeFileAdded.mock.calls[0][1]).toEqual({})
     })
 
+    it('should allow uploading duplicate file if explicitly allowed in onBeforeFileAdded', async () => {
+      const core = new Core({ onBeforeFileAdded: () => true })
+      const sameFileBlob = new File([sampleImage], { type: 'image/jpeg' })
+
+      core.addFile({
+        source: 'jest',
+        name: 'foo.jpg',
+        type: 'image/jpeg',
+        data: sameFileBlob,
+      })
+
+      await core.upload()
+
+      expect(() => {
+        core.addFile({
+          source: 'jest',
+          name: 'foo.jpg',
+          type: 'image/jpeg',
+          data: sameFileBlob,
+        })
+      }).not.toThrow()
```

Comment:
```suggestion
      core.addFile({
        source: 'jest',
        name: 'foo.jpg',
        type: 'image/jpeg',
        data: sameFileBlob,
      })
```

not.toThrow is implicit (an error thrown will always cause the test to fail)

also `await core.upload()` is irrelevant to this test